msal/src/main/java/com/microsoft/identity/client/NativeAuthPublicClientApplicationConfiguration.kt

```diff
@@ -0,0 +1,204 @@
+     * [NativeAuthPublicClientApplicationConfigurationFactory] cannot access it otherwise
+     */
+    public override fun validateConfiguration() {
+        // Check that a client id was passed
+        if (TextUtils.isEmpty(clientId)) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_USE_WITHOUT_CLIENT_ID_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_USE_WITHOUT_CLIENT_ID_ERROR_MESSAGE
+            )
+        }
+
+        // If we are given a redirectUri, we should run the superclass validation method to make sure
+        // we can use web auth
+        if (redirectUri != null) {
+            super.validateConfiguration()
+        } else {
+            LogSession.log(TAG, Logger.LogLevel.WARN, "No redirect URI was passed. You cannot use browser-based auth flow to handle BrowserRequired error.")
+        }
+
+        // Enforce that account mode must be "SINGLE"
+        if (accountMode != AccountMode.SINGLE) {
+            throw MsalClientException(
+                MsalClientException.NATIVE_AUTH_INVALID_ACCOUNT_MODE_CONFIG_ERROR_CODE,
+                MsalClientException.NATIVE_AUTH_INVALID_ACCOUNT_MODE_CONFIG_ERROR_MESSAGE
+            )
+        }
+
+        // We only allow one authority being passed
+        if (authorities == null || authorities.size != 1) {
+            // Throw is no authority passed
+            if (authorities == null || authorities.size == 0) {
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_USE_WITH_NO_AUTHORITY_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_USE_WITH_NO_AUTHORITY_ERROR_MESSAGE
+                )
+            } else {
+                // This throws when more than one authority is passed
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_USE_WITH_MULTI_AUTHORITY_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_USE_WITH_MULTI_AUTHORITY_ERROR_MESSAGE
+                )
+            }
+        }
+
+        if (defaultAuthority !is NativeAuthCIAMAuthority) {
+            // If the authority supplied is a CIAMAuthority, we can continue
+            if (defaultAuthority is CIAMAuthority) {
+                val nativeAuthAuthority = NativeAuthCIAMAuthority(
+                    authorityUrl = defaultAuthority.authorityUri.toString(),
+                    clientId = clientId
+                )
+                authorities.clear()
+                authorities.add(nativeAuthAuthority)
+            } else {
+                // If not, throw exception
+                throw MsalClientException(
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_CODE,
+                    MsalClientException.NATIVE_AUTH_INVALID_CIAM_AUTHORITY_ERROR_MESSAGE
+                )
+            }
+        }
+
+        // Enforce that shared device mode is disabled
+        if (isSharedDevice) {
```

Comment:
This is checking a value in the config JSON? or actually checking whether the device is in shared mode?